### ISSUE TITLE
Add `touch` to `Prelude`

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#5349]
Add:

```
def touch(_rc: Region[r]): Unit \ r = unsafe_cast () as Unit \ r
```

**Comment by paulbutcher** [comment]
Do we need this if we implement the machinery discussed [here](https://github.com/flix/flix/issues/4870#issuecomment-1396741690)?

**Comment by magnus-madsen** [comment]
> Do we need this if we implement the machinery discussed [here](https://github.com/flix/flix/issues/4870#issuecomment-1396741690)?

Yes. This is to overcome a limitation in the effect system.

**Comment by paulbutcher** [comment]
> > Do we need this if we implement the machinery discussed [here](https://github.com/flix/flix/issues/4870#issuecomment-1396741690)?
> 
> Yes. This is to overcome a limitation in the effect system.

For my information, what is that limitation?

**Comment by magnus-madsen** [comment]
From https://github.com/flix/flix/pull/5330,

We have:

```
pub def iterator(rc: Region[r1], s: MutSet[a, r2]): Iterator[a, r1 and r2, r1] \ { Write(r1), Read(r2) } =
  let MutSet(ms) = s;
  unsafe_cast Set.iterator(r1, deref ms) as Iterator[a, r1 and r2]
```

The problem is that `Set.iterator` has the specific type ` Iterator[a, r1, r1]`, but we really want it to have the type `Iterator[a, r1 and r2, r1] ` because we think that the iterator should "touch" region `r2` (the region of the mutable set).

However, because we have **cheated** and our mutable set is not really mutable at all, but rather is a pointer to an immutable set, Flix correctly infers that the iterator actually has no effect in `r2`. But to maintain the illusion that it does (and later allow for a more imperative implementation of `MutSet` we cast to ` Iterator[a, r1 and r2]`. But instead of casting, we can also get the right result if we just "touch" something in region `r2`. Hence this hack:

```
Set.iterator(rc, deref ms) |> Iterator.map(x -> {discard deref ms; x})
```

Now the iterator is clearly using `r2` because it dereferences `ms` _inside_ the iterator. 

This ticket just suggests to make that into a slightly more regular pattern, so you would write:

```
Set.iterator(rc, deref ms) |> Iterator.map(x -> {touch(...); x})
```

or something like that.

**Comment by magnus-madsen** [comment]
TLDR: Flix correctly infers that an expression is more pure than we want to pretend it is. To trick Flix, we can perform a no-op impure operations to get the right effect. To make this more systematic, we might as well introduce a `touch` function.

**Comment by paulbutcher** [comment]
OK, that makes sense. Thanks.

**Comment by magnus-madsen** [comment]
I think that the new iterator is very interesting. The effect system captures exactly what it means to be an iterator in the types, but it requires some new way of thinking about these things. Probably we should have a page on doc.flix.dev that explains.
